src/_P122_SHT2x.ino

```diff
@@ -0,0 +1,265 @@
+      }
+      break;
+    }
+
+    case PLUGIN_WEBFORM_LOAD:
+    {
+      // this case defines what should be displayed on the web form, when this plugin is selected
+      // The user's selection will be stored in
+      // PCONFIG(x) (custom configuration)
+
+      # define P122_RESOLUTION_OPTIONS 4
+
+      const __FlashStringHelper *options[P122_RESOLUTION_OPTIONS] = {
+        F("Temp 14 bits / RH 12 bits"),
+        F("Temp 13 bits / RH 10 bits"),
+        F("Temp 12 bits / RH  8 bits"),
+        F("Temp 11 bits / RH 11 bits"),
+      };
+      const int optionValues[P014_RESOLUTION_OPTIONS] = {
+        P122_RESOLUTION_14T_12RH,
+        P122_RESOLUTION_13T_10RH,
+        P122_RESOLUTION_12T_08RH,
+        P122_RESOLUTION_11T_11RH,
+      };
+      addFormSelector(F("Resolution"), P122_RESOLUTION_LABEL, P122_RESOLUTION_OPTIONS, options, optionValues, P122_RESOLUTION);
+
+      P122_data_struct *P122_data = static_cast<P122_data_struct *>(getPluginTaskData(event->TaskIndex));
+      if (P122_data != nullptr) 
+      {
+        uint32_t eida;
+        uint32_t eidb;
+        uint8_t firmware;
+        P122_data->getEID(&eida, &eidb, &firmware);
+        String txt = F("CHIP ID:");
+        txt += String(eida, HEX);
+        txt += F(",");
+        txt += String(eidb, HEX);
+        txt += F(" firmware=");
+        txt += String (firmware);
+        addFormNote(txt);
+      }
+
+      success = true;
+      break;
+    }
+
+    case PLUGIN_WEBFORM_SAVE:
+    {
+      // this case defines the code to be executed when the form is submitted
+      // the plugin settings should be saved to PCONFIG(x)
+      // ping configuration should be read from CONFIG_PIN1 and stored
+
+      // after the form has been saved successfuly, set success and break
+      P122_I2C_ADDRESS  = getFormItemInt(F("i2c_addr"));
+      P122_RESOLUTION   = getFormItemInt(P122_RESOLUTION_LABEL);
+        
+      P122_data_struct *P122_data = static_cast<P122_data_struct *>(getPluginTaskData(event->TaskIndex));
+      if (P122_data != nullptr) 
+      {
+        P122_data->setupDevice(P122_I2C_ADDRESS, P122_RESOLUTION);
+      }
+      success = true;
+      break;
+    }
+    case PLUGIN_INIT:
+    {
+      // this case defines code to be executed when the plugin is initialised
+      if (P122_I2C_ADDRESS == 0) 
+      {
+        P122_I2C_ADDRESS = P122_I2C_ADDRESS_AD0_0; // Use default address if not (yet) set
+      }
+      initPluginTaskData(event->TaskIndex, new (std::nothrow) P122_data_struct());
+      P122_data_struct *P122_data = static_cast<P122_data_struct *>(getPluginTaskData(event->TaskIndex));
+
+      if (P122_data != nullptr) 
+      {
+        P122_data->setupDevice(P122_I2C_ADDRESS, P122_RESOLUTION);
+        P122_data->reset();
+      }
+      UserVar[event->BaseVarIndex]     = NAN;
+      UserVar[event->BaseVarIndex + 1] = NAN;
+      UserVar[event->BaseVarIndex + 2] = NAN;
+      success = true;
```

Comment:
This should only be `true` if initialization actually is successful, so this assignment should be moved to be within the scope of the check a few lines up.